Sources/AppKitBackend/NSViewRepresentable.swift

```diff
@@ -0,0 +1,209 @@
+import SwiftCrossUI
+import AppKit
+
+public struct NSViewRepresentableContext<Coordinator> {
+    public let coordinator: Coordinator
+    public internal(set) var environment: EnvironmentValues
+}
+
+public protocol NSViewRepresentable: View
+where Content == Never {
+    associatedtype NSViewType: NSView
+    associatedtype Coordinator = Void
+
+    /// Create the initial NSView instance.
+    @MainActor
+    func makeNSView(context: NSViewRepresentableContext<Coordinator>) -> NSViewType
+
+    /// Update the view with new values.
+    /// - Parameters:
+    ///   - nsView: The view to update.
+    ///   - context: The context, including the coordinator and potentially new environment
+    ///   values.
+    /// - Note: This may be called even when `context` has not changed.
+    @MainActor
+    func updateNSView(_ nsView: NSViewType, context: NSViewRepresentableContext<Coordinator>)
+
+    /// Make the coordinator for this view.
+    ///
+    /// The coordinator is used when the view needs to communicate changes to the rest of
+    /// the view hierarchy (i.e. through bindings), and is often the view's delegate.
+    @MainActor
+    func makeCoordinator() -> Coordinator
+
+    /// Compute the view's size.
+    /// - Parameters:
+    ///   - proposal: The proposed frame for the view to render in.
+    ///   - nsVIew: The view being queried for its preferred size.
+    ///   - context: The context, including the coordinator and environment values.
+    /// - Returns: Information about the view's size. The ``SwiftCrossUI/ViewSize/size``
+    /// property is what frame the view will actually be rendered with if the current layout
+    /// pass is not a dry run, while the other properties are used to inform the layout engine
+    /// how big or small the view can be. The ``SwiftCrossUI/ViewSize/idealSize`` property
+    /// should not vary with the `proposal`, and should only depend on the view's contents.
+    /// Pass `nil` for the maximum width/height if the view has no maximum size (and therefore
+    /// may occupy the entire screen).
+    ///
+    /// The default implementation uses `nsView.intrinsicContentSize` and `nsView.sizeThatFits(_:)`
+    /// to determine the return value.
+    func determineViewSize(
```

Comment:
I feel like this should be `@MainActor` as well if the other uiview methods have to be